ocaml/xs-trace/xs_trace.ml

```diff
@@ -12,48 +12,93 @@
  * GNU Lesser General Public License for more details.
  *)
 
-let _ =
-  match Sys.argv with
-  | [|_; action; origin; url_string|] ->
-      let url = Uri.of_string url_string in
-      let submit_json json =
-        if json <> "" then
-          let result = Tracing_export.Destination.Http.export ~url json in
-          match result with
-          | Ok _ ->
-              ()
-          | Error err ->
-              Printf.eprintf "Error: %s" (Printexc.to_string err) ;
-              exit 1
-      in
-      let rec export_file orig =
-        if Sys.is_directory orig then
-          let files = Sys.readdir orig in
-          let file_paths = Array.map (Filename.concat orig) files in
-          Array.iter export_file file_paths
-        else if Filename.check_suffix orig ".zst" then
-          Xapi_stdext_unix.Unixext.with_file orig [O_RDONLY] 0o000
-          @@ fun compressed_file ->
-          Zstd.Fast.decompress_passive compressed_file @@ fun decompressed ->
-          if Filename.check_suffix orig ".ndjson.zst" then
+module Exporter = struct
+  module Unixext = Xapi_stdext_unix.Unixext
+
+  let ( >> ) f g x = g (f x)
+
+  (* Submit JSON to a specified endpoint. *)
+  let submit_json url json =
+    if json <> "" then
+      match Tracing_export.Destination.Http.export ~url json with
+      | Error err ->
+          Printf.eprintf "Error: %s" (Printexc.to_string err) ;
+          exit 1
+      | _ ->
+          ()
+
+  (** Export traces from file system to a remote endpoint. *)
+  let export erase src dst =
+    let dst = Uri.of_string dst in
+    let submit_json = submit_json dst in
+    let rec export_file = function
+      | path when Sys.is_directory path ->
+          (* Recursively export trace files. *)
+          Sys.readdir path |> Array.iter (Filename.concat path >> export_file)
```

Comment:
I feel that this is a little difficult to parse. The >> part is fine on its own (once I'd dusted off the part of my brain that recognised >> was just infix) but then combined with |> and Array.iter I think it's not as easily readable as the original.

I think if you wanted to keep it on one line, something like:
```suggestion
          Sys.readdir path |> Array.iter (fun file -> Filename.concat path file |> export_file)
```
would allow that (and you'd actually save a line because you wouldn't have to define >>